jsapp/js/components/formLanding.es6

```diff
@@ -388,6 +395,19 @@ export class FormLanding extends React.Component {
   setCollectMethod(evt) {
     this.setState({selectedCollectMethod: evt.currentTarget.dataset.method});
   }
+  goToProjectsList() {
+    hashHistory.push(ROUTES.FORMS);
+  }
+  nonOwnerSelfRemoval(evt) {
+    evt.preventDefault();
+    // Listen for permission removal here to avoid manage_asset user removal
+    // from triggering redirect
+    this.listenTo(
+      actions.permissions.removeAssetPermission.completed,
+      this.goToProjectsList
+    );
```

Comment:
I am wondering if you shouldn't remove the listener afterwards. So there is no bug after some other remove permission call :)

So:

```
nonOwnerSelfRemoval(evt) {
  evt.preventDefault();
  this.cancelNonOwnerSelfRemovalListener = this.listenTo(
    actions.permissions.removeAssetPermission.completed,
    this.nonOwnerSelfRemovalCompleted
  );
}
nonOwnerSelfRemovalCompleted() {
  if (this.cancelNonOwnerSelfRemovalListener) {
    this.cancelNonOwnerSelfRemovalListener()
  }
  this.goToProjectsList();
}
```

Reply:
I know it's not meant to be exact, but just to make sure can you please check my changes again :D